garak/_config.py

```diff
@@ -144,12 +146,16 @@ def _load_yaml_config(settings_filenames) -> dict:
 
 
 def _store_config(settings_files) -> None:
-    global system, run, plugins, reporting
+    global system, run, plugins, reporting, version
     settings = _load_yaml_config(settings_files)
     system = _set_settings(system, settings["system"])
     run = _set_settings(run, settings["run"])
+    run.user_agent = run.user_agent.replace("{version}", version)
     plugins = _set_settings(plugins, settings["plugins"])
     reporting = _set_settings(reporting, settings["reporting"])
+    from requests import utils
+
+    utils.default_user_agent = run.user_agent
```

Comment:
I like this single location for default. A question about long term viability. How can we ensure this only applies to `garak` objects?

Since the project wants to support inclusion as a dependency the first import of `garak._config` would override the `user_agent` for the current process until exit.  Should we consider this now or is this worth kicking the can down the road a bit?

Reply:
One pov is that child requests are still caused by garak anyway, and so it's in one way even desirable they adopt the UA

Reply:
Agreed, my concern is more with consumption as a library, the consuming process may do other `requests` after utilizing `garak`, these would also inherit the revised user-agent.

Consider an integration that runs various security tools in the same runtime and selects `garak` first, or also runs other benchmarks. I will offer I think the tool should spawn independent workspaces however if the consumer is already python it "may" not. In that scenario if the benchmark requests to not restore or reset the agent string requests could be attributed to the `garak` code when not related. Also if the tool expects to `post` results to some location further along in a CI pipeline this could introduce unexpected values in the headers for those calls.

Reply:
Excellent point. Maybe sans API we could push + pop UA as part of garak.cli for now.

Reply:
One idea for handling this might be to have this functionality injected and reset by `harnesses.base.Harness` when executing `run()` to limit the window of influence.

Reply:
IMO, having this within `garak.cli` is probably ideal -- we can use the same logic, but only set the default UA on invocation of the run.

Reply:
have placed it in `harnesses.base.Harness` for now, using two hooks and a class var, and covering `requests`, `httpx`, and `aiohttp`. Helper funcs staying in `_config`, which doesn't seem optimal, but might be least-bad

Reply:
added push & pop of user agent strings into `garak.cli` pending spec of api hook for doing whole run